### DISCUSSION TITLE
Improvements for command v new

### DISCUSSION THREAD

**Discussion by Zerotask** [vlang/v#11824] [Ideas]
The current `v new` command looks like this:
![image](https://user-images.githubusercontent.com/20150243/110017212-3f59a180-7d26-11eb-92d2-a4e1ae5c366f.png)

Suggestions:
~1. Option to define a project name directly: `v new test` (in this case the description would stay empty, no input asked)~ #9142
~2. Option to define a project name and description directly `v new test test123`~ #9142 
3. Add project types / templates (--type / --template, -t), e. g. `v new test --type web`. This would create a skeleton file with a very basic web server. There can be other types for android, ios etc
~4. Add a question about the version: `Input your version: (0.0.0) ` (since the created v.mod also have a version field)~ #9168
~5. Add a question about the license: `Input your license: (MIT) ` (V itself (and npm) is using a license field)~ #9168

in the JS world I'm used to the command `npm init -y` (stands for 'yes' to all questions) which will use all default options instead of getting asked. For the `name` the folder / directory name is used. This could also be an option for V: `v new -y`. #9179

<details>
<summary>npm init process (to compare it with the v new process)</summary>

See `npm help init` for definitive documentation on these fields
and exactly what they do.

Use `npm install <pkg>` afterwards to install a package and
save it as a dependency in the package.json file.

Press ^C at any time to quit.
package name: (test) v-test
version: (1.0.0)
description: this is a test
entry point: (index.js)
test command:
git repository:
keywords:
author:
license: (ISC)
About to write to C:\Users\Patrick\Dev\vlang\test\package.json:

{
  "name": "v-test",
  "version": "1.0.0",
  "description": "this is a test",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC"
}

Is this OK? (yes)
</details>

**Comment by Zerotask** [comment]
There could also be an argument for a dry run (--dry-run, -d). This wouldn't create anything but it would output the files it would create. This way users could check if this is what they want / expect. #9179

**Comment by spytheman** [comment]
I especially like the dry run idea - I often forget what `v init` vs `v create` does, and try them in empty folders just to see the difference.
